tests/topotests/bgp_route_map_match_source_protocol/test_bgp_route_map_match_source_protocol.py

```diff
@@ -109,6 +109,56 @@ def _bgp_check_advertised_routes_r3():
     _, result = topotest.run_and_expect(test_func, None, count=30, wait=1)
     assert result is None, "Failed to filter routes by source-protocol for r3"
 
+    def _bgp_check_advertised_routes_r4():
+        tgen.gears["r1"].vtysh_cmd(
```

Comment:
I'm a bit confused. You do the changes here, but you don't check anything, while below you do another change and then check for the results. Is that expected or how it works in general?

Reply:
here is what I wanted to achieve,

- After the previous test, (or to say the original .conf file) 1 Static and 3 connected paths
- I change configuration to use a policy r4(match protocol static) in redistribute static and connected and no match policy on a Nbr
- So my advertised route on that Nbr should only have one static route
- Restore config to original

Reply:
But can we do some additional real checks between configurations? Otherwise it looks odd.